### PR TITLE
Closes #138: Make `SCENARIO_URLS` dynamic and depending on the running test

### LINKED ISSUE #138: Create only necessary backstop reference during related test.
**Is your feature request related to a problem? Please describe.**
Right now, when we need to do VR testing we create references from the SCENARIO_URLS in wp.config irrespective of the test being run.

**Describe the solution you'd like**
It would make sense to conditionally create reference based on the current test being run, that way test run faster, less flaky on a poor internet connection with puppeteer trying to visit tons of urls and create reference and also we actually consume resource for only what we need.

I suggest we create a json or array file to hold various scenario urls related to their tests with their key as the script name e.g `test:llcssbg` this way we can get this script at run time and load it in `SCENARIO_URLS` in wp.config.ts just in time for backstop to create the references.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Miraeld**
# Description

Fixes #138
This PR introduces a change that allows scenario URLs to be loaded conditionally based on the current test being run. This improves the efficiency of our tests by only loading the URLs that are needed for each test, which can lead to faster test execution and less flakiness on poor internet connections

## Type of change

- [ ] New feature (non-breaking change which adds functionality).
- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] Enhancement (non-breaking change which improves an existing functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as before).
- [ ] Sub-task of #(issue number)
- [ ] Release

## Detailed scenario

To trigger the new code, simply run a test script. The scenario URLs for that specific test will be loaded from a JSON file and used in the test.

### Documentation

The code works by first importing a JSON file that maps script names to their respective scenario URLs. Then, in the `wp.config.ts` file, it retrieves the current script name using `process.env.npm_lifecycle_event`. This script name is used to look up the corresponding scenario URLs in the imported JSON object, and these URLs are assigned to `SCENARIO_URLS`.

### New dependencies

No new dependencies were introduced in this change.

### Risks

There are no known performance or security risks associated with this change. The main risk is that if the JSON file or the `wp.config.ts` file is not properly configured, the correct scenario URLs might not be loaded

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style
- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [x] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

# Additional Checks
- [x] In the case of complex code, I wrote comments to explain it.
- [x] When possible, I prepared ways to observe the implemented system (logs, data, etc.).
- [x] I added error handling logic when using functions that could throw errors (HTTP/API request, filesystem, etc.)

### CODE REVIEW

File: config/wp.config.sample.ts [hunk 1]
```diff
@@ -57,38 +59,14 @@ const {
...
-		mobile: true,
-	},
-	doubleColon: {
-		path: 'll_bg_css_double_colon'
-	},
-	singleColon: {
-		path: 'll_bg_css_single_colon'
-	}
+
+function getScenarioUrls(scriptName: string) {
```
Comment (jeawhanlee) [inline config/wp.config.sample.ts]: I'm thinking if we really need a function for this, in order to keep the config simple. Also should we not have a check to return default maybe like a ternary function when the script name is not found in the JSON?

**Comment by Mai-Saad** [comment]
@Miraeld @jeawhanlee Thanks for the PR. on PR running `test:smoke` will run backstop. can you please check :pray:

**Comment by jeawhanlee** [comment]
@Mai-Saad Do we have backstop references being created for urls with this PR, if not, I think we are good, what you mentioned will be fixed on a different issue [here](https://github.com/orgs/wp-media/projects/114/views/1?pane=issue&itemId=78909105)

File: config/wp.config.sample.ts (hunk 1 above)
Comment (Miraeld) [reply config/wp.config.sample.ts]: Yea we can remove it. done ;)

**Comment by Mai-Saad** [comment]
> Do we have backstop references being created for urls with this PR

yes
[screen-capture - 2024-09-19T134110.947.webm](https://github.com/user-attachments/assets/32022459-9225-4837-96f7-e0eea3f2ff3b)

**Comment by Miraeld** [comment]
@Mai-Saad Have you modified your own `wp.config.ts` to include the change related to `SCENARIO_URLS`. 
Because I don't get any reference for smoke tests, so I'm wondering ... 

You should have something like : 

``` typescript
import ScenarioUrls from './scenarioUrls.json';

// ... Rest of your config

const scriptName = process.env.npm_lifecycle_event;
const SCENARIO_URLS = ScenarioUrls[scriptName];

// ... Rest of your config

export {
    WP_USERNAME,
    WP_PASSWORD,
    WP_BASE_URL,
    WP_ROOT_DIR,
    WP_ENV_TYPE,
    WP_DOCKER_CONTAINER,
    WP_DOCKER_ROOT_DIR,
    WP_SSH_USERNAME,
    WP_SSH_ADDRESS,
    WP_SSH_KEY,
    WP_SSH_ROOT_DIR,
    SCENARIO_URLS
};

```

**Comment by Mai-Saad** [comment]
@Miraeld you are right, I was using the old config. updated the config and no references with smoke now :pray: and with llcssbg references are created

**Comment by jeawhanlee** [comment]
I can confirm this as well.
Screenshot 2024-09-19 at 12 12 31

**Review by jeawhanlee — APPROVED** [review]

**Comment by hanna-meda** [comment]
@Miraeld, @jeawhanlee, @Mai-Saad can confirm this, too. 

Tested also with `llimages`, `lcp`, `wpml` for which no references will be created ✓
Screenshot 2024-09-19 at 15 10 56

Backstop references will be created only for `llcssbg` ✓
Screenshot 2024-09-19 at 15 12 39

**Pull request merged**